src/components/product-option/product-option.tsx

```diff
@@ -6,18 +6,23 @@ import { getChoiceValue } from '~/utils';
 
 export interface ProductOptionProps {
     option: products.ProductOption;
+    allowedValues: Set<string> | undefined;
```

Comment:
this complicates the component.. let's better make sure that option contains only allowed choices

Reply:
then PR will change only 1 file 😄

Reply:
so then @VladyslavG would be very happy 😂